galaxy_ng/app/management/commands/update-synclist-distro-repos.py

```diff
@@ -0,0 +1,56 @@
+import logging
+
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+from galaxy_ng.app.models import SyncList
+
+log = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """This command modifies SyncLists repository to point to upstream_repository
+
+    Example:
+
+    django-admin update-synclist-distro-repos --point-each-to-upstream
+    django-admin update-synclist-distro-repos --point-each-to-synclist-repo
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument("--point-each-to-upstream",
+                            action="store_true",
+                            help="Update every SyncList's repository to point to the upstream repository (published, by default)",
+                            default=False,
+                            required=False,
+                            )
+        parser.add_argument("--point-each-to-synclist-repo",
+                            action="store_true",
+                            help="Update every SyncList's repository to point to the per group synclist ('12345-synclist' for example)",
+                            default=False,
+                            required=False,
+                            )
+
+    def handle(self, *args, **options):
+        log.debug('options: %s', repr(options))
+
+        synclists = SyncList.objects.all()
+
+        if options["point_each_to_upstream"]:
+            log.debug("Updating all synclists.repository to point to upstream_repository")
+
+            with transaction.atomic():
+                for synclist in synclists:
+                    log.debug('repo: %s upstream: %s', synclist.repository, synclist.upstream_repository)
+                    synclist.repository = synclist.upstream_repository
```

Comment:
Feels a little weird to be adding fields to synclist at this point. Won't that require migrations to run before the cmd?

If I recall correctly, I think there was some reason synclists didn't directly reference Distributions. ie, they just had matching names instead of a direct model attribute.

My memory on the details of the Repository->Distribution link are fuzzy though, I may be missing something.

Reply:
Yeah it would require a schema migration. We need the `galaxy_ng.models.Synclist` to be tied to the viewset, and since we will remove synclist repos, then distro is the only way to link them.

I'm not sure why we tied `galaxy_ng.models.Synclist` to the synclist repo and not distro... maybe since we would only need a synclist repo if we had a synclist obj to filter it. For the usage we expect now which is less complex, I think this is ok.